マルチモジュール化

imori425 opened the pull request:
# 対応内容

## 親のプロジェクトからバックエンドプロジェクトにコードを移行
親プロジェクトではソースコードを保持できません。
そのため、一時的にBackendプロジェクトに保持するように移行しました。

## mvn wrapper導入
メインクラスを実行できるようにするために導入しました。
うろ覚えなのですが、maven:wrapperで作成したファイルもバージョン管理に入れるべきだったと思います。正しいことは公式サイトで確認していただければと思います。

以下コマンドで導入できます。
```
mvn wrapper:wrapper
```


## spring-boot-maven-pluginにmainClassを指定
恐らく、jarファイルの中にあるメインクラスも見つかっていて、どちらを起動すればよいかmavenが判断できなかったのだと思われます。明示的に指定することで対応しました。
```
<plugin>
  <groupId>org.springframework.boot</groupId>
  <artifactId>spring-boot-maven-plugin</artifactId>
  <configuration>
    <mainClass>jp.yo41sawada.backend.Application</mainClass>
  </configuration>
</plugin>
```

## その他
コード上に記載しています。
わからない点がありましたら、質問をお願いいたします。

imori425 committed:
マルチモジュール化

imori425 commented on openapitools/.openapi-generator-ignore:
```diff
@@ -0,0 +1,28 @@
+#!docs/README.md
+
+
+# 手動で追記
+## 追加しないと上書きされてしまう
+pom.xml
```
pomを上書きされないように、ignoreに設定を追加しています。

imori425 commented on openapitools/pom.xml:
```diff
@@ -0,0 +1,66 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <parent>
+        <groupId>org.springframework.boot</groupId>
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>2.5.13</version>
+        <relativePath/>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <packaging>jar</packaging>
+    <artifactId>openapi-spring</artifactId>
+    <version>0.0.1-SNAPSHOT</version>
```
手動で変更を加えています。

imori425 commented on pom.xml:
```diff
@@ -1,12 +1,11 @@
-		<version>2.5.13</version>
-		<relativePath/> <!-- lookup parent from repository -->
-	</parent>
+	<packaging>pom</packaging>
```
親モジュールにするためにpomを設定しています。

yo41sawada replied:
こちらは想定通りなので OK です、ありがとうございます。

imori425 commented on pom.xml:
```diff
@@ -1,12 +1,11 @@
-		<relativePath/> <!-- lookup parent from repository -->
-	</parent>
+	<packaging>pom</packaging>
+	<modules>
+		<module>backend</module>
+		<module>openapitools</module>
+	</modules>
```
ここに子モジュールを追加します。

yo41sawada replied:
こちらは想定通りなので OK です、ありがとうございます。

imori425 commented on backend/src/main/java/jp/yo41sawada/backend/Application.java:
```diff
@@ -1,4 +1,4 @@
-package jp.yo41sawada.sample;
```
親モジュールにはソースコードを保持することができません。もともと親にあったコードはバックエンドに保持するように変更しています。

yo41sawada replied:
こちらは狙い通りなので OK です、ありがとうございます。

imori425 commented on pom.xml:
```diff
@@ -15,59 +14,22 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
 		<java.version>11</java.version>
 	</properties>
-	<dependencies>
```
親にソースコードを持てないため、ここの依存が不要になったので削除しています。

imori425 commented on pom.xml:
```diff
@@ -15,59 +14,22 @@
 				<groupId>org.openapitools</groupId>
 				<artifactId>openapi-generator-maven-plugin</artifactId>
 				<version>5.4.0</version>
+				<inherited>false</inherited>
```
子プロジェクトに設定が引き繋がれないようにfalseを設定しています。

imori425 commented on pom.xml:
```diff
@@ -15,59 +14,22 @@
 						</goals>
 						<configuration>
 							<inputSpec>${project.basedir}/openapi.yaml</inputSpec>
+							<output>${project.basedir}/openapitools</output>
```
openapiのgeneratorの出力先を設定しています。

yo41sawada replied:
こちらは想定通りなので OK です、ありがとうございます。

imori425 committed:
backendからopenapitoolsを参照できるように変更

imori425 committed:
Spring Boot Maven Pluginの導入方法を変更

https://spring.pleiades.io/spring-boot/docs/current/maven-plugin/reference/htmlsingle/#using.import

imori425 committed:
SpringBootのjarファイルを実行可能なjarファイルに変更

yo41sawada commented on openapitools/pom.xml:
```diff
@@ -0,0 +1,78 @@
+        </dependencies>
+    </dependencyManagement>
+
+    <dependencies>
```
ここから下の dependency が、当初分割前のものと変わっているようです。
main の方に私が検証したものを既にマージ済みでして、そちらと比較頂けるとイメージが付くかと思います。
